location/src/androidLibMain/kotlin/location/LocationMonitor.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ Copyright 2021 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.location
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.location.LocationManager
+import androidx.core.location.LocationManagerCompat
+import com.splendo.kaluga.base.ApplicationHolder
+import com.splendo.kaluga.base.ServiceMonitor
+
+actual class LocationMonitor(
+    private val applicationContext: Context,
+    private val locationManager: LocationManager?
+) : ServiceMonitor() {
+
+    actual class Builder {
+        actual fun create() = LocationMonitor(
+            applicationContext = ApplicationHolder.applicationContext,
+            locationManager = ApplicationHolder.applicationContext
+                .getSystemService(Context.LOCATION_SERVICE) as? LocationManager
```

Comment:
Same question, should we keep it optional and do checks inside isServiceEnabled or throw an error soon?

Reply:
This is different from bluetooth, in that case `BluetoothAdapter.getDefaultAdapter()` returns null when device does not support Bluetooth (like when you run from emulator). While here in location `getSystemService` takes a string and return a `Any` service for that and we need to cast it to the right service to access specific methods

Reply:
That does not mean it does not have to be optional, just add another `?` and an error.